libs/functions/generate_api.py

```diff
@@ -0,0 +1,25 @@
+from api.can_predictions import CANPredictionAPIRow, CANPredictionAPI, _Projections, _HospitalBeds
+from libs.datasets import results_schema as rc
+
+def _get_date_or_none(panda_date_or_none):
+    """ I think our null value is a string NULL so if this date value is a string,
+     make it none. Otherwise convert to the python datetime """
+    if isinstance(panda_date_or_none, str):
+        return None
+    return panda_date_or_none.to_pydatetime()
+
+def generate_api_for_projection(projection): 
+    api_results = []
+    # TODO: I think pandas doesn't really vibe with iterating, but it's 3000 and the other option is applying 
+    # things in a weird way so idk this is anti pandas but w/e 
+    for index, county_row in projection.iterrows(): 
+        peak_date = _get_date_or_none(county_row[rc.PEAK_HOSPITALIZATIONS])          
```

Comment:
When does this case happen?

Reply:
i added a comment for this it's basically our dates are null for this when there isn't a shortfall in beds